src/Lecture4.hs

```diff
@@ -237,7 +302,83 @@ you can return string "no value"
 -}
 
 displayStats :: Stats -> String
-displayStats = error "TODO"
+displayStats (Stats
+  (Sum statsTotalPositions)
+  (Sum statsTotalSum)
+  (Max statsAbsoluteMax)
+  (Min statsAbsoluteMin)   
+  Nothing
+  Nothing
+  (Just (Max statsBuyMax))       
+  (Just (Min statsBuyMin))        
+  (MaxLen statsLongest)) = unlines ["Total positions:       : " ++ show statsTotalPositions
+                                  , "Total final balance    : " ++ show statsTotalSum
+                                  , "Biggest absolute cost  : " ++ show statsAbsoluteMax
+                                  , "Smallest absolute cost : " ++ show statsAbsoluteMin
+                                  , "Max earning            : no value" 
+                                  , "Min earning            : no value"
```

Comment:
That's a lot of code duplication. You could write something like this to avoid handling of all 4 cases

```suggestion
                                  , "Max earning            : "  ++ maybe "no value" (show . getMax) statsSellMax
                                  , "Min earning            : "  ++ maybe "no value" (show . getMin) statsSellMin
```

And, most importantly, to avoid having `error` in the end